File: src/main/java/um/haberes/report/model/haberes/core/CargoClaseDetalleDto.java
package um.haberes.report.model.haberes.core;

import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CargoClaseDetalleDto {

    private Long cargoClaseDetalleId;

    private Long legajoId;

    private Integer anho = 0;

    private Integer mes = 0;

    private Long cargoClaseId;

    private Integer dependenciaId;

    private Integer facultadId;

    private Integer geograficaId;

    private int horas = 0;

    private BigDecimal valorHora = BigDecimal.ZERO;

    private Long cargoClasePeriodoId;

    private byte liquidado = 0;

    private PersonaDto persona;

    private CargoClaseDto cargoClase;

    private DependenciaDto dependencia;

    private FacultadDto facultad;

    private GeograficaDto geografica;

    private CargoClasePeriodoDto cargoClasePeriodo;

}

File: src/main/java/um/haberes/report/kotlin/dto/haberes/core/CargoClaseDetalleDto.kt
package um.haberes.report.kotlin.dto.haberes.core

import java.math.BigDecimal

data class CargoClaseDetalleDto(

    var cargoClaseDetalleId: Long? = null,
    var legajoId: Long? = null,
    var anho: Int? = 0,
    var mes: Int? = 0,
    var cargoClaseId: Long? = null,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var horas: Int = 0,
    var valorHora: BigDecimal = BigDecimal.ZERO,
    var cargoClasePeriodoId: Long? = null,
    var liquidado: Byte = 0,
    var persona: PersonaDto? = null,
    var cargoClase: CargoClaseDto? = null,
    var dependencia: DependenciaDto? = null,
    var facultad: FacultadDto? = null,
    var geografica: GeograficaDto? = null,
    var cargoClasePeriodo: CargoClasePeriodoDto? = null

)
